up-transport-vsomeip/tests/client_service.rs

```diff
@@ -0,0 +1,301 @@
+    }
+
+    async fn on_error(&self, err: UStatus) {
+        info!("{:?}", err);
+    }
+}
+pub struct RequestListener {
+    client: Weak<UPTransportVsomeip>,
+    received_request: AtomicUsize,
+}
+
+impl RequestListener {
+    #[allow(clippy::new_without_default)]
+    pub fn new(client: Arc<UPTransportVsomeip>) -> Self {
+        Self {
+            client: Arc::downgrade(&client),
+            received_request: AtomicUsize::new(0),
+        }
+    }
+
+    pub fn received_request(&self) -> usize {
+        self.received_request.load(Ordering::SeqCst)
+    }
+}
+
+#[async_trait::async_trait]
+impl UListener for RequestListener {
+    async fn on_receive(&self, msg: UMessage) {
+        self.received_request.fetch_add(1, Ordering::SeqCst);
+        info!("Received Request:\n{:?}", msg);
+
+        let payload = {
+            match msg.payload {
+                None => {
+                    panic!("Unable to retrieve bytes")
+                }
+                Some(payload) => payload,
+            }
+        };
+
+        let payload_bytes = payload.to_vec();
+        info!("Received request payload_bytes of: {payload_bytes:?}");
+        let Ok(payload_string) = std::str::from_utf8(&payload_bytes) else {
+            panic!("Unable to unpack string from payload_bytes");
+        };
+        info!("Request payload_string: {payload_string}");
+
+        let response_payload_string = format!("Here's a response to: {payload_string}");
+        let response_payload_bytes = response_payload_string.into_bytes();
+
+        let response_msg = UMessageBuilder::response_for_request(&msg.attributes)
+            .with_comm_status(UCode::OK.value())
+            .build_with_payload(response_payload_bytes, UPayloadFormat::UPAYLOAD_FORMAT_TEXT);
+        let Ok(response_msg) = response_msg else {
+            panic!(
+                "Unable to create response_msg: {:?}",
+                response_msg.err().unwrap()
+            );
+        };
+        if let Some(client) = self.client.upgrade() {
+            let send_res = client.send(response_msg).await;
+            if let Err(err) = send_res {
+                panic!("Unable to send response_msg: {:?}", err);
+            }
+        }
+    }
+
+    async fn on_error(&self, err: UStatus) {
+        info!("{:?}", err);
+    }
+}
+
+fn any_uuri() -> UUri {
+    UUri {
+        authority_name: "*".to_string(),
+        ue_id: 0x0000_FFFF,     // any instance, any service
+        ue_version_major: 0xFF, // any
+        resource_id: 0xFFFF,    // any
+        ..Default::default()
+    }
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn client_service() {
+    env_logger::init();
+    // console_subscriber::init();
+
+    let service_authority_name = "foo";
+    let streamer_ue_id = 0x7878;
+
+    let service_1_ue_id = 0x1234;
+    let service_1_ue_version_major = 1;
+    let service_1_resource_id_a = 0x0421;
+
+    let client_authority_name = "bar";
+    let client_ue_id = 0x0345;
```

Comment:
same over here... service_ue_id, client_ue_id and streamer_ue_id all 3 are different.... how do they communicate with each other ?

Reply:
Same kind of explanation as my comment [over here](https://github.com/eclipse-uprotocol/up-transport-vsomeip-rust/pull/4#discussion_r1698957576). The `streamer_ue_id` serves that purpose of the client_id of the vsomeip application that can be used to subscribe to publish messages.